### ISSUE TITLE
Cannot connect SDK to infobip server even though following all instructions

### ISSUE THREAD

**Issue by dameunbite** [infobip/mobile-messaging-react-native-plugin#32]
I have installed the sdk properly but cannot figure out how to connect it to infobip servers. I have been with this error for the past 2 weeks. What can I do to fix this issue?

# This is the error message:
`
{code: 1, domain: 'com.mobile-messaging', description: 'The application instance is not registered on the …ed to be notified as the registration succeeded.'}
code: 1
description: "The application instance is not registered on the server yet. The registration will be perfomed eventually. Subscribe to the notification `MMRegistrationUpdated` to be notified as the registration succeeded."
domain: "com.mobile-messaging"}
`

# Below is the code that is throwing the error.
`
      mobileMessaging.saveUser({
         externalUserId: 'randomid',
         emails: ['sampleemail@gmail.com'],
         phones: ['787XXXXXXX']
      }, user => {
        // Here you can use updated user instance
        console.log("saved user");
      }, (err) => {
        console.log(err);
      });
`

# Here are the images of the code and the aplication in the portal.

<img width="878" alt="Screen Shot 2021-12-06 at 4 57 45 PM" src="https://user-images.githubusercontent.com/48269003/144922132-4fda3029-3742-4560-895e-70c844c77b35.png">

![Screen Shot 2021-12-06 at 4 57 04 PM](https://user-images.githubusercontent.com/48269003/144922152-a2a43e8d-3fc8-487b-b3ab-91039964e7b1.png)

**Comment by riskpp** [comment]
Hi @dameunbite,

This error means that, registration wasn't performed yet and you need to receive "registrationUpdated" event, to be able to perform any other calls like "mobileMessaging.saveUser".

1. Don't you testing it on iOS Simulator just in case? On simulator we can't get APNS token, so it's not possible to register and test features.
2. You can check [Example application](https://github.com/infobip/mobile-messaging-react-native-plugin/wiki/How-to-start-Example-app), just in case you missed some integration part.
3. Recheck that you don't miss something from [Quick start guide](https://github.com/infobip/mobile-messaging-react-native-plugin#quick-start-guide).

BR, Olga

**Comment by dameunbite** [comment]
I have followed all the react-native integration and I keep getting the same error in production. This are the events logged from real device. 


mobileMessaging.init runs successfully and it does not give any error but on mobileMessaging.saveUser it gives the error displayed before (mobileMessaging.saveUser happens very quick after init). Is there another way we can troubleshoot this? Also can this error be more specific?


Should I listen to registrationUpdated first and then do mobileMessaging.saveUser after successfull registrationUpdated?


![Screen Shot 2021-12-08 at 4 47 20 PM](https://user-images.githubusercontent.com/48269003/145282071-6c18cf12-1f84-4fdd-9d93-e1a55568a3aa.png)
.

**Comment by riskpp** [comment]
Hi @dameunbite,

You can register to "registrationUpdated" event and make other calls after it happen.
Just to notice "registrationUpdated" event is posted during the registration process, after a current APNs device token is stored on the server and the new Push Registration Id for the current user is generated. So it'll be posted once, if device token already provided to the server this event won't come, talking easily after application restart this event won't come.

What else could be done to troubleshoot some other problems:
1. You could enable logging for iOS as in [Example](https://github.com/infobip/mobile-messaging-react-native-plugin/blob/master/Example/App.js#L114) and provide logs
2. You could register to all supported events as done [here](https://github.com/infobip/mobile-messaging-react-native-plugin/blob/master/Example/App.js#L92-L96) and put results to the console.log, just to check that registration went well.

BR, Olga

**Comment by dameunbite** [comment]
It's not working! I have added a log to our server to check what infobip package is responding. I get successful mobileMessaging.init callback but the app is not being reported in the server. Is even asking for the push notification and we are allowing it. 

Can we set up a meeting? 



![Screen Shot 2021-12-16 at 4 17 18 PM](https://user-images.githubusercontent.com/48269003/146443191-b8b56f6d-c5aa-4949-9305-e6a209edbc24.png)
<img width="846" alt="Screen Shot 2021-12-16 at 4 16 47 PM" src="https://user-images.githubusercontent.com/48269003/146443196-05d31471-061e-496a-abe0-3c4fbb315df4.png">
